src/Week1/CreditCard.hs

```diff
@@ -0,0 +1,72 @@
+module Week1.CreditCard where
+
+
+{-
+ Exercise 1: We need to first find the digits of a number.
+
+ 1. toDigits should convert positive Integers to a list of digits. (For 0 or
+ negative inputs, toDigits should return the empty list.)
+
+ 2. toDigitsRev should do the same, but with the digits reversed.
+
+ Examples: toDigits 1234 == [1,2,3,4]
+           toDigitsRev 1234 == [4,3,2,1]
+           toDigits 0 == []
+           toDigits (-17) == []
+ -}
+toDigits :: Integer -> [Integer]
+toDigits n
+  | n < 1     = []
+  | otherwise = toDigits (n `div` 10) ++ [n `mod` 10]
```

Comment:
The issue with this solution is that you use the `++` operator at every recursive call. Since the complexity of `++` is linear in the left list, you end up with an implementation that is O (n^2) in the length of the integer. 

This isn't necessarily an issue since the input integers would have to substantially large before it would affect performance. However, a more efficient solution would be preferable in this case, since it would reflect a more refined understanding of the problem itself.

 On that note, there are two possible efficient solutions. Here are the hints for them:
- One involves using a helper function that takes an additional accumulator list as an argument
- Another involves using the `:` operator and one application of `reverse`. 

Try implementing them!

Reply:
@oishikg I've implemented the above two functions. Can you see if that's what you meant?

Reply:
@nitinprakash96 yes indeed, you have implemented the functions that I expected